### PR TITLE
feat(argo-cd): Upgrade Argo CD to 2.6.0

### PR DESCRIPTION
**Pull request by pdrastil**
Upgrade to 2.6.0 stable version

Upgrade notes: https://argo-cd.readthedocs.io/en/latest/operator-manual/upgrading/2.5-2.6/

---

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/NOTES.txt
Comment (mtbennett-godaddy) [inline charts/argo-cd/templates/NOTES.txt:90]:
```suggestion
{{- if .Values.applicationSet.args.policy }}
DEPRECATED option applicationSet.args.policy - Use configs.params.applicationsetcontroller.policy
{{- end }}
{{- if .Values.applicationSet.args.dryRun }}
DEPRECATED option applicationSet.args.dryRun - Use configs.params.applicationsetcontroller.dryRun
{{- end }}
```

Comment (Lobstrosity) [inline charts/argo-cd/templates/NOTES.txt:90]:
```suggestion
{{- if .Values.applicationSet.args.policy }}
DEPRECATED option applicationSet.args.policy - Use configs.params.applicationsetcontroller.policy
{{- end }}
{{- if .Values.applicationSet.args.dryRun }}
DEPRECATED option applicationSet.args.dryRun - Use configs.params.applicationsetcontroller.dryRun
{{- end }}
```

Comment (pdrastil) [reply charts/argo-cd/templates/NOTES.txt]: Thanks fixed - good eye for copy & paste :)

**Review by jmeridth — APPROVED** [review]
LGTM 👍

**Comment by mbastawisy-careem** [comment]
Hey everyone, any updates on when this will be released?

**Comment by jmeridth** [comment]
@mbastawisy-careem once we get another approval.  Thank you for your patience.

**Review by yu-croco — APPROVED** [review]
LGTM! 🚀

File: charts/argo-cd/templates/crds/crd-applicationset.yaml
Comment (esphen) [inline charts/argo-cd/templates/crds/crd-applicationset.yaml:1473]: Is it intentional that this is required now? I'm getting the following error when upgrading:

```
failed to apply ApplicationSet.argoproj.io, argocd/skip-apps: ApplicationSet.argoproj.io "skip-apps" is invalid: spec.generators[0].git.         pathParamPrefix: Required value
```

The documentation seems to infer that it should be optional: 
https://argo-cd.readthedocs.io/en/release-2.6/operator-manual/applicationset/Generators-Git/#git-generator-directories

Comment (pdrastil) [reply charts/argo-cd/templates/crds/crd-applicationset.yaml:1473]: Good catch @esphen - missed that one with diff. PR contains the fix https://github.com/argoproj/argo-helm/pull/1829

Comment (bennesp) [reply charts/argo-cd/templates/crds/crd-applicationset.yaml:1473]: ASAIK it was an upstream error, but solved in v2.6.0-rc3 here: https://github.com/argoproj/argo-cd/pull/11943

So probably these CRDs are not up-to-date?

**Pull request merged**
